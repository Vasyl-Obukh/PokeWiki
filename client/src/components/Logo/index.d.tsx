import Logo from './index';

declare module Logo {}