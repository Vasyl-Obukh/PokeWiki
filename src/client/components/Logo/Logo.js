import React from 'react';
import { Link } from 'react-router-dom';
import styled from 'styled-components';
import paths from '../../constants/paths';

const LogoLink = styled(Link)`

`;

const LogoImage = styled.img`
  width: 144px;
  height: 66px;
`;

const Logo = () => {
  return (
    <LogoLink to={paths.HOME}>
      <LogoImage src='../../assets/images/logo.png' alt='logo'/>
    </LogoLink>
  );
};

export default Logo;