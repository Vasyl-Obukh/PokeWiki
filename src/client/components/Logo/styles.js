import styled from 'styled-components';
import {Link} from 'react-router-dom';

export const StyledLink = styled(Link)`

`;

export const Image = styled.img`
  width: 144px;
  height: 66px;
`;