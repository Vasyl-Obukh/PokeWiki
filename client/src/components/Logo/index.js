import React from 'react';
import PropTypes from 'prop-types';
import * as Styles from './styles';
import { reset } from 'redux-form';
import { connect } from 'react-redux';
import paths from '../../constants/paths';

export const Logo = ({ resetForm }) => {
  return (
    <Styles.StyledLink to={paths.HOME} onClick={resetForm}>
      <Styles.Image src='../../static/images/logo.png' alt='logo'/>
    </Styles.StyledLink>
  );
};

Logo.propTypes = {
  resetForm: PropTypes.func
};

export const mapDispatchToProps = dispatch => ({
  resetForm: () => dispatch(reset('filter'))
});

export default connect(
  null,
  mapDispatchToProps
)(Logo);