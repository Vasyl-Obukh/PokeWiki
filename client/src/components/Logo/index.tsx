import * as React from 'react';
import * as Styles from './styles';
import { reset } from 'redux-form';
import { connect } from 'react-redux';
import paths from '../../constants/paths';

type DispatchProps = {
  resetForm: () => void
}

type Props = DispatchProps; // | StateProps

export const Logo = (props: Props) => {
  const { resetForm } = props;

  return (
    <Styles.StyledLink to={paths.HOME} onClick={resetForm}>
      <Styles.Image src='../../static/images/logo.png' alt='logo'/>
    </Styles.StyledLink>
  );
};

export const mapDispatchToProps = dispatch => ({
  resetForm: () => dispatch(reset('filter'))
});

export default connect(
  null,
  mapDispatchToProps
)(Logo);