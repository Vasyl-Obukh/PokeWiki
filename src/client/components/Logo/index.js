import React from 'react';
import * as Styles from './styles';
import paths from '../../constants/paths';

const Logo = () => {
  return (
    <Styles.StyledLink to={paths.HOME}>
      <Styles.Image src='../../static/images/logo.png' alt='logo'/>
    </Styles.StyledLink>
  );
};

export default Logo;