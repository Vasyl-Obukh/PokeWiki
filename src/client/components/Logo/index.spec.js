import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
import Logo from './index';

Enzyme.configure({adapter: new Adapter()});

describe('<Logo /> tests', () => {
  const mockStore = configureStore();
  const initialState = {};
  const store = mockStore(initialState);

  test('Should render correctly', () => {
    const component = renderer.create(
      <Provider store={store}>
        <Router>
          <Logo />
        </Router>
      </Provider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Should render image from predefined directory', () => {
    const component = mount(
      <Provider store={store}>
        <Router>
          <Logo />
        </Router>
      </Provider>
    );
    expect(component.find('img').prop('src')).toContain('static/images/logo.png');
  });
});