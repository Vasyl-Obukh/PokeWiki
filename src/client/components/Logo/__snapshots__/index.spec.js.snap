// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> tests Should render correctly 1`] = `
<a
  className="sc-bdVaJa eAwGjB"
  href="/"
  onClick={[Function]}
>
  <img
    alt="logo"
    className="sc-bwzfXH cAQUzw"
    src="../../static/images/logo.png"
  />
</a>
`;
